src/application_config/configuring_base.py

```diff
@@ -21,49 +22,45 @@
 
 
 @dataclass(frozen=True)
-class ConfigSectionBase(ABC):
-    """Base class for all ConfigSection classes"""
+class ConfigSectionBase:
+    """Base class for all ConfigSection classes (so that we can bound a TypeVar)"""
 
 
-# The next line has a type: ignore because MyPy has a problem with @abstractmethod
-# If you comment the line with @abstractmethod, you can do type checking
 @dataclass(frozen=True)
-class ConfigBase(ABC):
+class ConfigBase:
     """Base class for main Config class"""
 
-    @staticmethod
-    @abstractmethod
-    def get_app_basename() -> str:
-        """Return the application base name.
-
-        This name, with a preceding dot, will also be the folder name in the home directory
-        that will store the config.
-        """
+    @classmethod
+    def default_config_foldername(cls: type[TConfig]) -> str:
+        """Return the class name without 'Config' and with a preceding dot as default for the folder that will hold the config file."""
+        return f".{cls.__name__.replace('Config', '')}"
 
     @classmethod
-    def get_configfile_path(cls: type[TConfig]) -> Path:
-        """Return the fully qualified path for the configfile: e.g. ~/.stockwatch/stockwatch-config.toml"""
-        return Path.home() / f".{cls.get_app_basename()}" / "config.toml"
+    def defaut_config_filepath(cls: type[TConfig]) -> Path:
+        """Return the fully qualified path for the configfile: e.g. ~/.Example/config.toml"""
+        return Path.home() / f"{cls.default_config_foldername()}" / "config.toml"
 
     @classmethod
-    def get(cls: type[TConfig], reload: bool = False) -> TConfig:
+    def get(
+        cls: type[TConfig], reload: bool = False, configfile_path: str = ""
```

Comment:
`configfile_path` should probably be a `None | Path`

Reply:
(Is present at more locations)

Reply:
I deliberately chose this string interface. It means less to code on client side and an empty string evaluates to False in conditions. So in my view the current design is more user friendly.

Reply:
I deliberately chose this string interface. It means less to code on client side and an empty string evaluates to False in conditions. So in my view the current design is more user friendly. Can you agree?

Reply:
Okay I have some very strong opinions about usage of strings, and they can be summarized by being **only** used for human readable representation of something. This is largely driven by a 'rule' from functional languages, "Making illegal states unrepresentable" see for example [this (older) blog](https://fsharpforfunandprofit.com/posts/designing-with-types-making-illegal-states-unrepresentable/). A string is usually just a cop out for putting any data in (e.g. `\n😃:/` is a perfectly valid string, but not a valid path on any OS.

There is also the example if I want to make a python app for Windows and Linux. The `Path` object nicely handles all OS specifics, and ensures I create valid paths for both, so any such application will already (or should already) use Path objects internally, then transform them to strings for this method, which recreates the Path object again?

So in the end I think we should use a Path, because it has to be a Path :-).

Reply:
I now provide to use either a string or a Path. And I validate the string.